components/localized_input/localized_input.tsx

```diff
@@ -2,55 +2,41 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import {intlShape} from 'react-intl';
+import {injectIntl, IntlShape} from 'react-intl';
 
 type Props = {
     placeholder: {
         id: string;
         defaultMessage: string;
     };
     value?: string;
-}
-
-export default class LocalizedInput extends React.Component<Props> {
-    public static contextTypes = {
-        intl: intlShape.isRequired,
-    };
-    public input: React.RefObject<HTMLInputElement> = React.createRef<HTMLInputElement>();
-
-    public get value(): string {
-        return this.input.current ? this.input.current.value : '';
-    }
-
-    public set value(value: string) {
-        if (this.input.current) {
-            this.input.current.value = value;
-        }
-    }
-
-    public focus = (): void => {
-        if (this.input.current) {
-            this.input.current.focus();
-        }
-    };
+    intl: IntlShape;
+    forwardedRef?: React.RefObject<HTMLInputElement>;
+};
 
+class LocalizedInput extends React.Component<Props> {
     public shouldComponentUpdate(nextProps: Props): boolean {
         return nextProps.value !== this.props.value ||
             nextProps.placeholder.id !== this.props.placeholder.id ||
             nextProps.placeholder.defaultMessage !== this.props.placeholder.defaultMessage;
     }
 
     public render(): JSX.Element {
-        const {formatMessage} = this.context.intl;
-        const {placeholder, ...otherProps} = this.props;
+        const {formatMessage} = this.props.intl;
+        const {placeholder, forwardedRef, ...otherProps} = this.props;
         const placeholderString: string = formatMessage(placeholder);
 
         return (
             <input
-                ref={this.input}
                 {...otherProps}
+                ref={forwardedRef}
                 placeholder={placeholderString}
             />
         );
     }
-}
\ No newline at end of file
+}
+
+const Component = injectIntl(LocalizedInput, {forwardRef: true});
+Component.displayName = 'forwardRef(injectIntl(input))';
```

Comment:
displayName is inferred from the `.name` property in JavaScript. This means that if react-intl use an anonymous function or an anonymous class inside, it can be lost.

This is common case in ref forwarding components. See: https://ko.reactjs.org/docs/forwarding-refs.html#displaying-a-custom-name-in-devtools